packages/ember-data/lib/system/references/has-many.js

```diff
@@ -0,0 +1,84 @@
+import Ember from 'ember';
+import Reference from './reference';
+
+const get = Ember.get;
+
+var HasManyReference = function(store, parentInternalModel, hasManyRelationship) {
+  this._super$constructor(store, parentInternalModel);
+  this.hasManyRelationship = hasManyRelationship;
+  this.type = hasManyRelationship.relationshipMeta.type;
+
+  if (hasManyRelationship.link) {
+    this.remoteType = "link";
+    this.link = hasManyRelationship.link;
+  } else {
+    this.remoteType = "ids";
+    this._ids = hasManyRelationship.canonicalState.map(function(internalModel) {
```

Comment:
canonicalState is server state. What we care about is members(or current state but members is better for now). But do not do this eagerly, it's slow and unneccesary